documentation/reference/search-definitions-reference.html

```diff
@@ -2644,20 +2637,29 @@ <h2 id="field_types">Field types</h2>
 
 <tr><th id="type:reference">reference&lt;document-type&gt;</th>
 <td>
+<p>
 A <em>reference&lt;document-type&gt;</em> field is a reference to an instance of a document-type -
-i.e. a foreign key. The reference is the <a href="../documents.html">document id</a>
-of the document-type instance, hence a string.
-References are used to join documents in a
-<a href="../search-definitions.html#document-references">parent-child relationship</a>.
+i.e. a foreign key.
+The reference is the <a href="../documents.html">document id</a> of the document-type instance, hence a string attribute.
+References are used to join documents in a <a href="../parent-child.html">parent-child relationship</a>.
 A reference can only be made to <a href="services-content.html#document">global</a> documents.
+</p><p>
+The following type of references are not supported:
+<ul>
+  <li>Self-reference</li>
+  <li>Cyclic reference: If document type <em>foo</em> has a reference to <em>bar</em>,
+    then <em>bar</em> cannot have a reference to <em>foo</em></li>
+</ul>
+</p>
 <table class="table">
   <thead></thead><tbody>
     <tr>
       <th style="width:100px">Indexing</th>
       <td>Invalid - deployment will fail.</td>
     </tr><tr>
       <th>Attribute</th>
-      <td>As <a href="#type:string">string</a> - a reference must be an attribute.</td>
+      <td>As <a href="#type:string">string</a> - a reference must be an attribute.
+          Can be NULL or point to a non-existing document</td>
```

Comment:
I think NULL can be misleading here. Consider replacing with "empty string" or "not specified"

Reply:
agreed!